workflow/worker.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2024 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package workflow
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"reflect"
+	"runtime"
+	"strings"
+	"sync"
+
+	dapr "github.com/dapr/go-sdk/client"
+
+	"github.com/microsoft/durabletask-go/backend"
+	durabletaskclient "github.com/microsoft/durabletask-go/client"
+	"github.com/microsoft/durabletask-go/task"
+)
+
+type WorkflowWorker struct {
+	tasks  *task.TaskRegistry
+	client *durabletaskclient.TaskHubGrpcClient
+
+	mutex  sync.Mutex // TODO: implement
+	quit   chan bool
+	close  func()
+	cancel context.CancelFunc
+}
+
+type Workflow func(ctx *WorkflowContext) (any, error)
+
+type Activity func(ctx ActivityContext) (any, error)
+
+func NewWorker() (*WorkflowWorker, error) {
```

Comment:
can you make it so a worker can be initialized using an existing dapr client? I just noticed looking at the example included in this PR that it would initialize 3 different dapr clients, one for the worker, another for just using it and another for the workflow client... and I was wondering if all of that could just be reduced to a single dapr client

Reply:
Added an option so that a dapr client can be passed to create the worker.

Reply:
thank you, would it make sense to do the same to the same in `workflow/client.go` ? and btw, just out of curiosity, why `workflow/client.go` doesn't it serve the same purpose as the management functions added here `client/client.go`?